runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java

```diff
@@ -105,7 +105,14 @@ public DataflowUnboundedReadFromBoundedSource(BoundedSource<T> source) {
 
   @Override
   public String getKindString() {
-    return "Read(" + approximateSimpleName(source.getClass()) + ")";
+    String sourceName;
+    if (source.getClass().isAnonymousClass()) {
+      sourceName = "AnonymousSource";
+    } else {
+      sourceName = approximateSimpleName(source.getClass());
+    }
+
+    return "Read(" + sourceName + ")";
```

Comment:
`approximateSimpleName(Class<?>)` [fails on anonymous classes](https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/util/StringUtils.java#L209), so this change won't affect anything that was previously working. This change matches the [default `PTransform` implementation](https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/PTransform.java#L242).

I hit the issue during this change because my test was sending an anonymous `Source` class into the transform. I could avoid it by using a concrete class, but no reason to step around an existing bug.